lib/improver/tests/ensemble_calibration/ensemble_calibration/test_EstimateCoefficientsForEnsembleCalibration.py

```diff
@@ -676,105 +650,148 @@ def test_coefficient_values_for_truncated_gaussian_distribution(self):
         expected values for a truncated Gaussian distribution."""
-        data = [3.16843498e-06, -5.34489037e-06,
-                9.99985648e-01, 1.00000028e+00]
+        distribution = "truncated gaussian"
+
+        plugin = Plugin(distribution, self.current_cycle)
+        result = plugin.estimate_coefficients_for_ngr(
+            self.historic_wind_speed_forecast_cube,
+            self.wind_speed_truth_cube)
+
+        self.assertArrayAlmostEqualLowerPrecision(
+            result.data, self.expected_mean_predictor_truncated_gaussian)
+        self.assertArrayEqual(
+            result.coord("coefficient_name").points, self.coeff_names)
 
+    @ManageWarnings(
+        ignored_messages=IGNORED_MESSAGES, warning_types=WARNING_TYPES)
+    def test_coefficients_truncated_gaussian_default_initial_guess(self):
+        """Ensure that the values for the optimised_coefficients match the
+        expected values, and the coefficient names also match
+        expected values for a truncated Gaussian distribution."""
+        expected = [0., 1.5434, -0.5141, 0.94]
         distribution = "truncated gaussian"
-        current_cycle = "20171110T0000Z"
 
-        plugin = Plugin(distribution, current_cycle)
+        plugin = Plugin(distribution, self.current_cycle)
+        plugin.ESTIMATE_COEFFICIENTS_FROM_LINEAR_MODEL_FLAG = False
         result = plugin.estimate_coefficients_for_ngr(
             self.historic_wind_speed_forecast_cube,
             self.wind_speed_truth_cube)
 
-        self.assertArrayAlmostEqual(result.data, data)
+        self.assertArrayAlmostEqualLowerPrecision(result.data, expected)
         self.assertArrayEqual(
             result.coord("coefficient_name").points, self.coeff_names)
 
+    @unittest.skipIf(
+        STATSMODELS_FOUND is False, "statsmodels module not available.")
     @ManageWarnings(
         ignored_messages=IGNORED_MESSAGES, warning_types=WARNING_TYPES)
-    def test_coefficient_values_for_gaussian_distribution_realizations(self):
+    def test_coefficients_gaussian_realizations_statsmodels(self):
         """Ensure that the values for the optimised_coefficients match the
         expected values, and the coefficient names also match
         expected values for a Gaussian distribution where the
         realizations are used as the predictor of the mean."""
-        import imp
-        try:
-            imp.find_module('statsmodels')
-            statsmodels_found = True
-        except ImportError:
-            statsmodels_found = False
-
-        if statsmodels_found:
-            data = [-0.00114, -0.00006, 1.00037, -0.00196, 0.99999, -0.00315]
-        else:
-            data = [4.30804737e-02, 1.39042785e+00, 8.99047025e-04,
-                    2.02661310e-01, 9.27197381e-01, 3.17407626e-01]
-
-        distribution = "gaussian"
-        current_cycle = "20171110T0000Z"
-        desired_units = "Celsius"
+        data = [-0.0003, 1.0023, -0.2831, -0.0774, 0.3893, 0.9168]
```

Comment:
I've tried to group together the results from these different tests, so that they can be more easily compared side-by-side, as you've suggested. I've also modified the precision to which the results are stated.

Reply:
Thanks, this is much clearer now.